crates/swc_ecma_transforms_typescript/src/inline_enum.rs

```diff
@@ -0,0 +1,135 @@
+use std::{cell::RefCell, rc::Rc};
+
+use serde::{Deserialize, Serialize};
+use swc_atoms::JsWord;
+use swc_common::collections::AHashMap;
+use swc_ecma_ast::{
+    AssignExpr, ComputedPropName, Expr, Id, Lit, MemberExpr, MemberProp, Module, Script, Str, Tpl,
+};
+use swc_ecma_utils::ident::IdentLike;
+use swc_ecma_visit::{as_folder, noop_visit_mut_type, Fold, VisitMut, VisitMutWith};
+
+pub type TSEnumLit = Rc<RefCell<AHashMap<Id, AHashMap<JsWord, Lit>>>>;
+
+pub fn inline_enum(ts_enum_lit: TSEnumLit, inline_enum: InlineEnumConfig) -> impl Fold + VisitMut {
+    as_folder(InlineEnum {
+        ts_enum_lit,
+        inline_enum,
+    })
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum EnumConst {
+    Const,
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+pub enum InlineEnumConfig {
+    Const(EnumConst),
+    All(bool),
+}
+
+impl Default for InlineEnumConfig {
+    fn default() -> Self {
+        Self::All(true)
```

Comment:
If so, we have two options I guess.

The first one is adding a field to `Assumptions` and the second one is defaulting to `EnumConst::Const`.
I think the second one would be better at the moment

Reply:
I prefer an assumption, we need to expose `Assumptions` in a near future anyway.

Reply:
Oh, I see. I think the assumption is good too.

Reply:
That would also need to change the options from `InlineEnumConfig` to `bool`